src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -644,50 +654,41 @@ internal override FlowAnalysisAnnotations FlowAnalysisAnnotations
-                    !_packedFlags.TryGetWellKnownAttribute(assertsFalse, out bool hasAssertsFalse))
+                FlowAnalysisAnnotations value;
+                if (!_packedFlags.TryGetFlowAnalysisAnnotations(out value))
                 {
-                    FlowAnalysisAnnotations? annotations = TryGetExtraAttributeAnnotations();
-
-                    if (annotations.HasValue)
-                    {
-                        // External annotations win, if any is present on the member
-                        hasNotNullWhenTrue = (annotations & FlowAnalysisAnnotations.NotNullWhenTrue) != 0;
-                        hasNotNullWhenFalse = (annotations & FlowAnalysisAnnotations.NotNullWhenFalse) != 0;
-                        hasAssertsTrue = (annotations & FlowAnalysisAnnotations.AssertsTrue) != 0;
-                        hasAssertsFalse = (annotations & FlowAnalysisAnnotations.AssertsFalse) != 0;
-                    }
-                    else
-                    {
-                        bool hasEnsuresNotNull = _moduleSymbol.Module.HasAttribute(_handle, AttributeDescription.EnsuresNotNullAttribute);
-                        hasNotNullWhenTrue = hasEnsuresNotNull || _moduleSymbol.Module.HasAttribute(_handle, AttributeDescription.NotNullWhenTrueAttribute);
-                        hasNotNullWhenFalse = hasEnsuresNotNull || _moduleSymbol.Module.HasAttribute(_handle, AttributeDescription.NotNullWhenFalseAttribute);
-                        hasAssertsTrue = _moduleSymbol.Module.HasAttribute(_handle, AttributeDescription.AssertsTrueAttribute);
-                        hasAssertsFalse = _moduleSymbol.Module.HasAttribute(_handle, AttributeDescription.AssertsFalseAttribute);
-                    }
-
-                    _packedFlags.SetWellKnownAttribute(notNullWhenTrue, hasNotNullWhenTrue);
-                    _packedFlags.SetWellKnownAttribute(notNullWhenFalse, hasNotNullWhenFalse);
-                    _packedFlags.SetWellKnownAttribute(assertsTrue, hasAssertsTrue);
-                    _packedFlags.SetWellKnownAttribute(assertsFalse, hasAssertsFalse);
-
-                    if (annotations.HasValue)
-                    {
-                        return annotations.Value;
-                    }
+                    // External annotations win, if any is present on the member
+                    value = TryGetExtraAttributeAnnotations() ?? DecodeFlowAnalysisAttributes(_moduleSymbol.Module, _handle);
+                    _packedFlags.SetFlowAnalysisAnnotations(value);
                 }
+                return value;
+            }
+        }
 
-                return FlowAnalysisAnnotationsFacts.Create(
-                    notNullWhenTrue: hasNotNullWhenTrue, notNullWhenFalse: hasNotNullWhenFalse,
-                    assertsTrue: hasAssertsTrue, assertsFalse: hasAssertsFalse);
+        private static FlowAnalysisAnnotations DecodeFlowAnalysisAttributes(PEModule module, ParameterHandle handle)
```

Comment:
>PEModule module, ParameterHandle handle [](start = 76, length = 39)

Is there an advantage to pass these as parameters?

Reply:
The method can be `static`.

---
In reply to: [286263041](https://github.com/dotnet/roslyn/pull/35846#discussion_r286263041) [](ancestors = 286263041)

Reply:
> The method can be static.

And what is the advantage of that? There is only one call site and it is in an instance method.

---
In reply to: [286306009](https://github.com/dotnet/roslyn/pull/35846#discussion_r286306009) [](ancestors = 286306009,286263041)